ibm/resource_ibm_private_dns_glb_monitor.go

```diff
@@ -0,0 +1,520 @@
+				Description: "The timeout (in seconds) before marking the health check as failed",
+			},
+
+			pdnsGlbMonitorMethod: {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validateAllowedStringValue([]string{"GET", "HEAD"}),
+				Description:  "The method to use for the health check",
+			},
+
+			pdnsGlbMonitorPath: {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The endpoint path to health check against",
+			},
+
+			pdnsGlbMonitorHeaders: {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						pdnsGlbMonitorHeadersName: {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+
+						pdnsGlbMonitorHeadersValue: {
+							Type:     schema.TypeList,
+							Required: true,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+						},
+					},
+				},
+			},
+
+			pdnsGlbMonitorAllowInsecure: {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "Do not validate the certificate when monitor use HTTPS. This parameter is currently only valid for HTTPS monitors.",
+			},
+
+			pdnsGlbMonitorExpectedCodes: {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validateAllowedStringValue([]string{"200", "201", "202", "203", "204", "205", "206", "207", "208", "226", "2xx"}),
+				Description:  "The expected HTTP response code or code range of the health check. This parameter is only valid for HTTP and HTTPS",
+			},
+
+			pdnsGlbMonitorExpectedBody: {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "A case-insensitive sub-string to look for in the response body",
+			},
+
+			pdnsGlbMonitorCreatedOn: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "GLB Monitor creation date",
+			},
+
+			pdnsGlbMonitorModifiedOn: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "GLB Monitor Modification date",
+			},
+		},
+	}
+}
+
+func resourceIBMPrivateDNSGLBMonitorCreate(d *schema.ResourceData, meta interface{}) error {
+	sess, err := meta.(ClientSession).PrivateDnsClientSession()
+	if err != nil {
+		return err
+	}
+	instanceID := d.Get(pdnsInstanceID).(string)
+	CreateMonitorOptions := sess.NewCreateMonitorOptions(instanceID)
+
+	monitorname := d.Get(pdnsGlbMonitorName).(string)
+	monitorinterval := int64(d.Get(pdnsGlbMonitorInterval).(int))
+	monitorretries := int64(d.Get(pdnsGlbMonitorRetries).(int))
+	monitortimeout := int64(d.Get(pdnsGlbMonitorTimeout).(int))
+	var monitortype string
+	var monitorport int64
+
+	if monitordescription, ok := d.GetOk(pdnsGlbMonitorDescription); ok {
+		CreateMonitorOptions.SetDescription(monitordescription.(string))
+	}
+	if Mtype, ok := d.GetOk(pdnsGlbMonitorType); ok {
+		monitortype = Mtype.(string)
+	} else {
+		monitortype = "HTTP"
+	}
+
+	if monitortype == "HTTP" {
```

Comment:
```
if Mport, ok := d.GetOk(pdnsGlbMonitorPort); ok {
 			monitorport = int64(Mport.(int))
 		}
```
This check would be enough..
Lets not hardcode values for `monitorport ` here.. 
API generally handles it..

Reply:
Done